ClosedXML/Excel/CalcEngine/Functions/Tally.cs

```diff
@@ -56,85 +61,87 @@ public void Add(Expression e)
             else
                 foreach (var v in valEnumerable)
                     _list.Add(v);
+
+            _numericValues = null;
         }
 
         public void AddValue(Object v)
         {
             _list.Add(v);
+            _numericValues = null;
         }
 
         public double Count()
         {
-            return this.Count(this.NumbersOnly);
+            return Count(NumbersOnly);
         }
 
         public double Count(bool numbersOnly)
         {
             if (numbersOnly)
-                return NumericValues().Count();
+                return NumericValues().Count;
             else
-                return _list.Where(o => !Statistical.IsBlank(o)).Count();
+                return _list.Count(o => !Statistical.IsBlank(o));
         }
 
-        public IEnumerable<Double> NumericValues()
+        IEnumerable<double> NumericValuesInternal()
         {
-            var retVal = new List<double>();
             foreach (var value in _list)
             {
-                Double tmp;
                 var vEnumerable = value as IEnumerable;
-                if (vEnumerable == null && Double.TryParse(value.ToString(), out tmp))
-                    yield return tmp;
+                if (vEnumerable == null)
+                {
+                    if (double.TryParse(value.ToString(), out double tmp))
```

Comment:
I had issues with this part of code too. On heavy formulas performing ```double.TryParse(doubleValue.ToString(), out tmp)``` took 50% of entire CPU usage being absolutely unnecessary.  See https://github.com/ClosedXML/ClosedXML/pull/679/files#diff-dcbb29e3ee481081b1cd4cd877f6a3a0R84